core/src/test/java/de/danoeh/antennapod/core/storage/ItemEnqueuePositionCalculatorTest.java

```diff
@@ -0,0 +1,279 @@
+                    {"download order test, enqueue after currently playing",
+                            list(11L, 101L, 12L, 13L, 14L),
+                            list(11L, 101L, 102L, 12L, 13L, 14L),
+                            AFTER_CURRENTLY_PLAYING, QUEUE_DEFAULT, 11L}  // 11 is at the front, currently playing
+            });
+        }
+
+        @Parameter
+        public String message;
+
+        @Parameter(1)
+        public List<Long> idsExpectedAfter101;
+
+        @Parameter(2)
+        public List<Long> idsExpectedAfter102;
+
+        @Parameter(3)
+        public EnqueueLocation options;
+
+        @Parameter(4)
+        public List<FeedItem> queueInitial;
+
+        @Parameter(5)
+        public long idCurrentlyPlaying;
+
+        @Test
+        public void testQueueOrderWhenDownloading2Items() {
+
+            // Setup class under test
+            //
+            ItemEnqueuePositionCalculator calculator = new ItemEnqueuePositionCalculator(options);
+            DownloadStateProvider stubDownloadStateProvider = mock(DownloadStateProvider.class);
+            stub(stubDownloadStateProvider.isDownloadingFile(any(FeedMedia.class))).toReturn(false);
+            calculator.downloadStateProvider = stubDownloadStateProvider;
+
+            // Setup initial data
+            // A shallow copy, as the test code will manipulate the queue
+            List<FeedItem> queue = new ArrayList<>(queueInitial);
+
+
+            // Test body
+
+            Playable currentlyPlaying = getCurrentlyPlaying(idCurrentlyPlaying);
+
+            // User clicks download on feed item 101
+            FeedItem tFI101 = setAsDownloading(101, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (1st download)",
+                    calculator, tFI101, queue, currentlyPlaying,
+                    idsExpectedAfter101);
+
+            // Then user clicks download on feed item 102
+            FeedItem tFI102 = setAsDownloading(102, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (2nd download, it should preserve order of download)",
+                    calculator, tFI102, queue, currentlyPlaying,
+                    idsExpectedAfter102);
+
+            // TODO: simulate download failure cases.
+        }
+
+
+        private static FeedItem setAsDownloading(int id, DownloadStateProvider stubDownloadStateProvider) {
+            FeedItem item = createFeedItem(id);
+            FeedMedia media =
+                    new FeedMedia(item, "http://download.url.net/" + id
+                            , 100000 + id, "audio/mp3");
+            media.setId(item.getId());
+            item.setMedia(media);
+
+            stub(stubDownloadStateProvider.isDownloadingFile(media)).toReturn(true);
+
+            return item;
+        }
+    }
+
+
+    // Common helpers:
+    // - common queue (of items) for tests
+    // - construct FeedItems for tests
```

Comment:
Shouldn't this be Javadoc?

Reply:
They were applied to the set of static members / methods. I've removed the comment as it's quite evident by the names.